### ISSUE TITLE
upload gets stuck when "parallel" on Windows

### ISSUE THREAD

**Issue by yarikoptic** [dandi/dandi-cli#1323]
this might be a duplicate of a ghost (I can't see it - says doesn't exit) issue https://github.com/dandi/dandi-cli/issues/1321 by @ZhuJiaBCM (also according to github - non existing account for me but does provide hyperlink and details with name and location to confirm my sanity... filed a support issue).  From an email correspondence:

>    As I told in slack, the upload succeeded by running :
>    'set DANDI_DEVEL=1', then 'dandi -l DEBUG upload --devel-debug'
>    The later trials for 'dandi -l DEBUG upload --J 1:1' and 'dandi -l DEBUG
>    upload --J 1:5' worked, 
>    'dandi -l DEBUG upload --J 5:1' and 'dandi -l DEBUG upload --J 5:5' got
>    stuck.
>    As Yarik mentioned, the default uploading is 5 files at the same time-- 5
>    treads in parallel. So, I guess the reason I failed in previous uploading
>    is it can't deal multiple files at the same time. 

and processes were stuck in `pre-validating` state.  Might be something about caching by fscacher or locking... heh

**Comment by CodyCBakerPhD** [comment]
#1321 is now visible
